airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/GcsStorageOperations.java

```diff
@@ -25,29 +24,13 @@ public GcsStorageOperations(final NamingConventionTransformer nameTransformer,
   }
 
   /**
-   * This method is overridden because GCS doesn't accept request to delete multiple objects. The
-   * only difference is that the AmazonS3#deleteObjects method is replaced with
-   * AmazonS3#deleteObject.
+   * This method is overridden because GCS doesn't accept request to delete multiple objects. The only
+   * difference is that the AmazonS3#deleteObjects method is replaced with AmazonS3#deleteObject.
    */
   @Override
-  public void cleanUpBucketObject(final String objectPath, final List<String> stagedFiles) {
-    final String bucket = s3Config.getBucketName();
-    ObjectListing objects = s3Client.listObjects(bucket, objectPath);
-    while (objects.getObjectSummaries().size() > 0) {
-      final List<KeyVersion> keysToDelete = objects.getObjectSummaries()
-          .stream()
-          .map(obj -> new KeyVersion(obj.getKey()))
-          .filter(obj -> stagedFiles.isEmpty() || stagedFiles.contains(obj.getKey()))
-          .toList();
-      for (final KeyVersion keyToDelete : keysToDelete) {
-        s3Client.deleteObject(bucket, keyToDelete.getKey());
-      }
-      LOGGER.info("Storage bucket {} has been cleaned-up ({} objects were deleted)...", objectPath, keysToDelete.size());
-      if (objects.isTruncated()) {
-        objects = s3Client.listNextBatchOfObjects(objects);
-      } else {
-        break;
-      }
+  protected void cleanUpObjects(final String bucket, final List<KeyVersion> keysToDelete) {
+    for (final KeyVersion keyToDelete : keysToDelete) {
+      s3Client.deleteObject(bucket, keyToDelete.getKey());
```

Comment:
huh why is the s3Client being used in the GCS class?

Reply:
It's normal for the moment because we use s3client to interact with GCS:
https://github.com/airbytehq/airbyte/blob/2f850b98ac1ba11aef00974642dbffd03fbf8ce8/airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/GcsDestinationConfig.java#L22